src/include/migraphx/op/batch_norm.hpp

```diff
@@ -42,7 +42,7 @@ struct batch_norm_inference
     shape compute_shape(std::vector<shape> inputs) const
     {
         check_shapes{inputs, *this}.has(5);
-        check_shapes{inputs.data(), inputs.data() + 1, *this}.only_dims(4);
+        check_shapes{inputs.data(), inputs.data() + 1, *this}.same_ndims();
```

Comment:
You should update the gpu operator to check for 4 dims.

Reply:
Should it be 4 dims on the GPU? I thought MIOpen can support up to 3D batchnorm?

Reply:
But for now, I dont think we call the API correctly for 3d convolutions.